Renaming appsec page + a few updates (#6358)
fleetdm/fleet: docs/Using-Fleet/Application-security.md

### Describe your secure coding practices, including code reviews, use of static/dynamic security testing tools, 3rd party scans/reviews.

Code commits to Fleet go through a series of tests, including SAST (static application security
testing). We use a combination of tools including [gosec](https://github.com/securego/gosec) and
[CodeQL](https://codeql.github.com/) for this purpose.

> Every piece of code that is merged into Fleet is reviewed by at least one other engineer before

Reviewer: ```suggestion
At least one other engineer reviews every piece of code before it is merged to Fleet.
```

> merging.  This is enforced via branch protection on the main branch.

Reviewer: ```suggestion
Merging,  we enforce this via branch protection on the main branch.
```

Author: Not sure why Merging with an uppercase `M`? It's at the beginning of a line but not a sentence, possibly a wrapping issue @Desmi-Dizney?

Reviewer: @GuillaumeRoss, if that's the case I'm confused because it already says "merged" in this sentence. Perhaps a rewrite for clarity would be helpful.

Author: @Desmi-Dizney rewrote! Can I haz re-review? :)

### Components with known vulnerabilities – prevent the use of libraries, frameworks, other software with existing vulnerabilities.

> We rely on Github's automated vulnerability checks, community news, and direct reports to discover

Reviewer: ```suggestion
We rely on GitHub's automated vulnerability checks, community news, and direct reports to discover
```
